diff --git a/src/src/tcmalloc.cc b/src/src/tcmalloc.cc
index 819d5a9..e2bc0b3 100644
--- a/src/tcmalloc.cc
+++ b/src/tcmalloc.cc
@@ -1137,7 +1137,10 @@ TCMallocGuard::TCMallocGuard() {
   }
 
 #ifndef WIN32_OVERRIDE_ALLOCATORS
-  ReplaceSystemAlloc();    // defined in libc_override_*.h
+  const char* env = getenv("TCMALLOC_REPLACE_SYSTEM_ALLOC_DISABLE");
+  if (env == nullptr || *env == 0 || stricmp(env, "true") != 0) {
+    ReplaceSystemAlloc();    // defined in libc_override_*.h
+  }
   (void)MallocExtension::instance(); // make sure malloc extension is constructed
   tc_free(tc_malloc(1));
 #endif  // !WIN32_OVERRIDE_ALLOCATORS
